src/particles/ImpactXParticleContainer.cpp

```diff
@@ -107,6 +108,49 @@ namespace impactx
         SetParticleShape(v);
     }
 
+    void
+    ImpactXParticleContainer::prepare ()
+    {
+        // make sure we have at least one box with enough tiles for each OpenMP thread
+
+        // make sure level 0, grid 0 exists
+        int lid = 0, gid = 0;
+        {
+            const auto& pmap = ParticleDistributionMap(lid).ProcessorMap();
+            auto it = std::find(pmap.begin(), pmap.end(), amrex::ParallelDescriptor::MyProc());
+            if (it == std::end(pmap)) {
+                amrex::Abort("Particle container needs to have at least one grid.");
+            } else {
+                gid = *it;
+            }
+        }
+
+        int nthreads = 1;
+#if defined(AMREX_USE_OMP)
+        nthreads = omp_get_max_threads();
+#endif
+
+        const auto& ba = ParticleBoxArray(lid);
+        auto n_logical = numTilesInBox(ba[gid], true, tile_size);
+
+        int ntry = 0;
+        while ((n_logical < nthreads) && (ntry++ < 6)) {
+            int idim = (ntry % 2) + 1;  // alternate between 1 and 2
+            tile_size[idim] /= 2;
+            n_logical = numTilesInBox(ba[gid], true, tile_size);
+        }
+
+        if (n_logical < nthreads) {
+            amrex::Abort("ImpactParticleContainer::prepare() "
```

Comment:
@atmyers I added more more concrete guidance to users here, can you please double check it?

In which situations would we not be able to find enough tiles? Is there other guidance we should give?

Reply:
Currently, the code will not just be efficient, it will give incorrect results if the number of tiles according to the `tile_size` is fewer than the number of cells. The issue is that the copyParticles routine in AMReX does not copy the ones that aren't on a valid tile. This resulted in some of the particles not getting written out some examples (since we us copyParticles to a pinned_pc in IO). 

This can be changed in AMReX, I'd suggest changing this to a warning rather than an Abort at that point.